File: cs50/pset1/mario.c
#include <cs50.h>
#include <stdio.h>

// Read user input
int get_input()
{
    int n;

    do
    {
        n = get_int("Height: ");
    } while ((n <= 0) || (n > 8));

    return n;
}

// Print the pyramid
void print_pyramid(int n)
{
    for (int i = 0; i < n; i++)
    {
        // Print first half
        for (int j = 0; j < n; j++)
            (j < (n - i - 1)) ? printf(" ") : printf("#");

        // Print two spaces
        printf("  ");

        // Print second half
        for (int j = 0; j < n; j++)
            if (j <= i)
                printf("#");

        // Print new line
        printf("\n");
    }
}

int main(void)
{
    int n = get_input();
    print_pyramid(n);
    exit(0);
}

File: cs50/pset6/mario/mario.py
n = 0
while True:
    try:
        n = int(input("Height:"))
    except ValueError:
        continue

    if 0 < n < 9:
        break

for i in range(1, n + 1):
    print((n - i) * " " + i * "#" + "  " + i * "#")
